astropy/io/fits/tests/test_connect.py

```diff
@@ -653,3 +654,61 @@ def test_info_attributes_with_no_mixins(tmpdir):
     assert t2['col0'].description == 'hello' * 40
     assert t2['col0'].format == '{:8.4f}'
     assert t2['col0'].meta['a'] == {'b': 'c'}
+
+
+@pytest.mark.skipif('not HAS_YAML')
+def test_round_trip_masked_table_default(tmpdir):
+    """This is a regression test showing the currently BROKEN default
+    behavior of Table.write() to a FITS bintable.  Note:
+
+    >>> simple_table(masked=True)
+    <Table masked=True length=3>
+      a      b     c
+    int64 float64 str1
+    ----- ------- ----
+       --     1.0    c
+        2     2.0   --
+        3      --    e
+    """
+    filename = str(tmpdir.join('test.fits'))
+
+    t = simple_table(masked=True)
+    t['a'].fill_value = 2
+    t.write(filename)
+    t2 = Table.read(filename)
+
+    assert t2.masked is True
+    assert np.all(t2['a'].mask == [True, True, False])  # This is WRONG
+    assert np.all(t2['b'].mask == [False, False, False])  # This is WRONG
+    assert np.all(t2['c'].mask == [False, False, False])  # This is WRONG
+
+    assert np.all(t2['a'] == [1, 2, 3])
+    assert np.all(t2['b'][:2] == [1.0, 2.0])  # OK
+    assert np.isnan(t2['b'][2])  # WRONG
+    assert np.all(t2['c'] == ['c', 'N', 'e'])  # WRONG ('N' is first char of 'N/A')
```

Comment:
I think I'd rather have an xfail test with the right asserts than a test with the wrong asserts.

Reply:
In the end I think it makes more sense to take this "failure regression" test out and open an issue that can be tracked.  XFAILing tests is not the most sensible / useful way to track bugs.

Reply:
Sounds good; just be sure there is a link to that issue in the test file.